rust/enums/src/main.rs

```diff
@@ -0,0 +1,205 @@
+use std::fs;
+use std::path::Path;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, PartialEq)]
+enum AgeBracket {
+    Child,
+    Youth,
+    Adult,
+    Senior,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct Person {
+    id: Option<u32>,
+    age: Option<u32>,
+    is_married: Option<bool>,
+    city: Option<String>,
+    state: Option<String>,
+    country: Option<String>,
+    age_bracket: Option<AgeBracket>,
+}
+
+impl Person {
+    fn set_age_bracket(&mut self) {
+        self.age_bracket = match self.age {
+            Some(a) => {
+                if a < 13 {
+                    Some(AgeBracket::Child)
+                } else if (13..=17).contains(&a) {
+                    Some(AgeBracket::Youth)
+                } else if (18..=59).contains(&a) {
+                    Some(AgeBracket::Adult)
+                } else {
+                    Some(AgeBracket::Senior)
+                }
+            }
+            None => None,
+        };
+    }
+}
+
+fn load_csv(file_path: &Path) -> Result<Vec<Person>, Box<dyn std::error::Error>> {
+    let contents = fs::read_to_string(file_path).expect("Unable to read from CSV");
+    let mut reader = csv::Reader::from_reader(contents.as_bytes());
+    let mut data: Vec<Person> = reader.deserialize().collect::<Result<_, _>>()?;
+
+    for d in &mut data {
+        d.set_age_bracket();
+    }
+
+    Ok(data)
+}
+
+fn main() {
+    let file_path = Path::new("data/persons.csv");
+    let persons = load_csv(file_path).expect("Unable to read/open CSV");
+    for person in persons {
+        println!("{:?}", person);
+    }
+}
+
+#[cfg(test)]
```

Comment:
I think it makes sense to separate the tests into their own file `test_main.rs`, even though Rust allows keeping them in the same file. It just keeps things consistent with the Python way of thinking, and also helps scale to larger projects when using modules in Rust.

Reply:
My reason for doing this is it is the standard Rust way of working, so if someone is working on a Rust project this is how they will find the unit tests. That said, I don't have a strong opinion here so I'll split them out.

Reply:
Just to confirm: in the standard Rust way for larger projects, would each file's tests be self contained within the file itself? If so, maybe then I should consider changing the approach in the other projects too.

Reply:
I don't have too strong an opinion against the standard way of doing things in any language, lol 😂

Reply:
What I have seen in the projects I have worked on/looked at is the unit tests like we have here go into the file itself, and integration tests that cover multiple pieces/modules go into a `tests` directory like in Python. As far as I understand it this is similar to all the tests in a Python project going into the `tests` directory. The test runner doesn't care where you put it, it has just become a convention people tend to follow.

Reply:
Ahh, no wonder I got confused when reading some example Rust code bases then. They must have been integration tests. Feel free to keep this as is, I'll go in and change the other tests to be consistent then.

Reply:
That's an excellent point! I think I can make this point via a callout block in the book via a subsection so that the difference is clear. 👍🏽